File: 프로그래머스/0/181882. 조건에 맞게 수열 변환하기 1/조건에 맞게 수열 변환하기 1.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer;
    for(int i: arr) {
        if (i>=50) {
            if(i%2==0)
                answer.push_back(i/2);
            else answer.push_back(i);
        }
        if(i<50) {
            if(i%2!=0) answer.push_back(i*2);
            else answer.push_back(i);
        }
    }
    return answer;
}

File: 프로그래머스/0/181882. 조건에 맞게 수열 변환하기 1/조건에 맞게 수열 변환하기 1.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr) {
        int[] answer = {};
        List<Integer> li = new ArrayList<>();
        for(int i=0; i<arr.length; i++) {
            if (arr[i]>=50 && arr[i]%2==0) {
                li.add(arr[i]/2);
            } else if (arr[i]<50 && arr[i]%2!=0) {
                li.add(arr[i]*2);
            } else li.add(arr[i]);
        }
        answer = new int[li.size()];
        for(int i=0; i<li.size(); i++) {
            answer[i] = li.get(i);
        }
        return answer;
    }
}